src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package duke;
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Deadline class
+ */
+public class Deadline extends Task {
+    protected LocalDate byDate;
+    protected String byString;
+
+    /** 
+     * Constructor for Deadline
+     * 
+     * @param description
+     * @param by
+     * @throws DateTimeException
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.byString = by;
+        try {
+            this.byDate = parseDate(by, f1, f2, f3, f4); // Parse the by string
+        } catch (DateTimeException e) {
+            throw new DateTimeException("Error parsing date in Deadline: " + e.getMessage());
+        }
+    }
+
+    @Override
+    protected String taskType() {
```

Comment:
We should use verbs for method names in accordance to the [CS2103 Java Coding Standard](https://se-education.org/guides/conventions/java/intermediate.html#naming). I noticed this issue elsewhere too.

```suggestion
    protected String getTaskType() {
```